draft-ietf-quic-transport.md

```diff
@@ -6057,12 +6057,15 @@ DecodePacketNumber(largest_pn, truncated_pn, pn_nbits):
    // The following code calculates a candidate value and
    // makes sure it's within the packet number window.
    candidate_pn = (expected_pn & ~pn_mask) | truncated_pn
-   if candidate_pn <= expected_pn - pn_hwin:
+   // Note the extra check for overflow when candidate_pn
+   // is near MAX_PACKET_NUMBER
+   if candidate_pn <= expected_pn - pn_hwin and
+      candidate_pn + pn_win < 1 << 62:
```

Comment:
If the intent is to avoid overflow, then it might be a good idea to avoid overflow on intermediate values also:

```suggestion
      candidate_pn < (1 << 62) - pn_win:
```

Reply:
Yes, agreed